src/pybindings/pycontroller/create_error_wrapper.py

```diff
@@ -0,0 +1,150 @@
+import sys
+import re
+import os
+import argparse
+
+
+class ErrorCode(object):
+    def __init__(self, error_name, error_code):
+        self.name = error_name.replace('_', ' ').title().replace(' ', '')
+        self.message = error_name
+        self.code = int(error_code, 16)
+
+    def __str__(self):
+        return '<ErrorCode name="{}" code="{}">'.format(self.name, self.code)
+
+    def __repr__(self):
+        return '<ErrorCode name="{}" code={}>'.format(self.name, self.code)
+
+    def create_exception_class(self):
+        return '''
+struct <name> : public std::exception
+{
+    const char * what () const throw ()
+    {
+    	return "<message>";
+    }
+};
+'''.replace('<name>', self.name).replace('<message>', self.message)
+
+    def create_py_bind(self, module_name):
+        return 'py::register_exception<chip::PythonBindings::{}>({}, "{}");'.format(self.name, module_name, self.name)
+
+    def create_else_if(self, var_name):
+        return '''
+    else if (<var_name> == <error_code>) {
+        throw chip::PythonBindings::<name>();
+    }'''.replace('<var_name>', var_name).replace('<error_code>', str(self.code)).replace('<name>', self.name)
+
+
+def generate_header(error_codes):
+    structs = "".join([x.create_exception_class() for x in error_codes])
```

Comment:
Can you comment on why this is required? Imho we could have a single exception class for CHIP errors and expose `ErrorStr` to python for human readable errors.

Reply:
This is just to use the easy Exception feature of Pybind.  Pybind automatically converts C++ exceptions to Python ones.  We had a requirement from @andreilitvin  to make sure that Python did exceptions even though the sdk returns error codes.  This was a quick and easy way to translate all of those to exceptions without writing a lot of extra code.

Reply:
@kidip But we could easily have just one exception and put the numeric error code on it as a property and avoid the need for code generation.

Reply:
Yeah, this was mostly for translating to the human readable name. I am not bound to any approach, this just seemed like the easiest one at the time.

Reply:
I think we should, separate types is frankly inconsistent with C++ which uses a single type.

An exception can have a message, which can include the message corresponding to the error code.

Reply:
Assuming we keep the goal of outputting a human readable error message we'd still have to map codes to a human readable string correct? So, even with 1 class we'd need some generation in order to not hardcode the messages?  Or do you think that we should hard code the messages and keep them up to date with any changes of errors?  (All this is under the assumption that the errors will change)

Reply:
Could we use `chip::ErrorStr` function at the time of creating an exception object instead of creating the whole mapping in advance?

Reply:
For exceptions you have to bind the class via PyBind in advance, but could have code that calls that error string method (which I didn't know existed ;) )

Reply:
We can definitely bind chip::ErrorStr, but will also have to bind chip::RegisterCHIPLayerErrorFormatter to get the chip formatted string. I think since it involves binding of targeting the actual SDK, we can do this in a separate PR.  This does not raise exceptions, but we can make a class that uses this to raise to appropriate exception based off of the error string. This is how it would eventually look to obtain the error string:

Python 3.8.2+chromium.10 (7b3ab5921f-dirty:7b3ab5921f, Feb 25 2020, 04:20:01) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pychip.PyChip import chip
>>> chip.RegisterCHIPLayerErrorFormatter()
>>> chip.ErrorStr(50)
'CHIP Error 0x00000032: Timeout'

Reply:
I think best bet is to get this in, then we can refactor the Exceptions.  These are mostly used on the Python side of things anyways, so easy to change.